storey/flow.py

```diff
@@ -1134,6 +1134,14 @@ async def _terminate(self):
 
     async def _do(self, event):
         if event is _termination_obj:
+            # Cancel _timeout_task to prevent memory leak (ML-11518)
```

Comment:
It's good to cancel it, but it doesn't seem like it would be the cause of a leak, as the task would just time out. I would be more worried that it could affect the next run – i.e. cause a logical bug rather than a leak. As such, I doubt it relates to ML-11518.